src/cppnv/EnvReader.cc

```diff
@@ -0,0 +1,813 @@
+    }
+  }
+  switch (key_char) {
+    case '`':
+      if (value->back_tick_quoted) {
+        return false;
+      }
+      add_to_buffer(value, key_char);
+      break;
+    case '#':
+      if (value->implicit_double_quote) {
+        return false;
+      }
+      add_to_buffer(value, key_char);
+      break;
+    case '\n':
+      if (!(value->triple_double_quoted || value->triple_quoted || (value->
+              double_quoted && !value->implicit_double_quote))) {
+        if (value->value_index > 0) {
+          if (value->value->at(value->value_index - 1) == '\r') {
+            value->value_index--;
+          }
+        }
+        return false;
+      }
+      add_to_buffer(value, key_char);
+      return true;
+    case '\\':
+
+      if (value->quoted || value->triple_quoted) {
+        add_to_buffer(value, key_char);
+        return true;
+      }
+      value->back_slash_streak++;
+
+      return true;
+    case '{':
+      add_to_buffer(value, key_char);
+      if (!value->quoted && !value->triple_quoted) {
+        if (!value->is_parsing_variable) {
+          // check to see if it's an escaped '{'
+          if (!is_previous_char_an_escape(value)) {
+            open_variable(value);
+          }
+        }
+      }
+      return true;
+    case '}':
+      add_to_buffer(value, key_char);
+      if (value->is_parsing_variable) {
+        // check to see if it's an escaped '}'
+        if (!is_previous_char_an_escape(value)) {
+          close_variable(value);
+        }
+      }
+
+      return true;
+    case '\'':
+      if (!value->double_quoted && !value->triple_double_quoted) {
+        value->single_quote_streak++;
+      } else {
+        add_to_buffer(value, key_char);
+      }
+      return true;
+
+    case '"':
+      if (!value->quoted && !value->triple_quoted && !value->back_tick_quoted &&
+          !value->implicit_double_quote) {
+        value->double_quote_streak++;
+      } else {
+        add_to_buffer(value, key_char);
+      }
+      return true;
+
+    default:
+      add_to_buffer(value, key_char);
+  }
+  return true;
+}
+
+// Used only when checking closed and open variables because the { }
+// // have been added to the buffer
+// // it needs to check 2 values back.
+bool EnvReader::is_previous_char_an_escape(const EnvValue* value) {
+  return value->value_index > 1
+         && value->value->at(value->value_index - 2) ==
+         '\\';
+}
+
+
+EnvReader::read_result EnvReader::read_value(EnvStream* file,
+                                             EnvValue* value) {
+  if (!file->good()) {
+    return end_of_stream_value;
+  }
+
+  char key_char = 0;
+  while (file->good()) {
+    key_char = file->get();
+    if (key_char < 0) {
+      break;
+    }
+    // if (clear_newline_or_comment(file, value, key_char, ret_val))
+    //   break;
+    if (read_next_char(value, key_char) && file->good()) {
+      continue;
+    }
+    // if (!(value->triple_double_quoted || value->triple_quoted)) {
```

Comment:
this is also an unused method, we can remove this

Reply:
Please make this clearer?

Reply:
Same over here https://github.com/nodejs/node/pull/50835#discussion_r1404866000